src/lib.rs

```diff
@@ -440,7 +440,13 @@ pub trait Strategy<'a> {
 /// A strategy that uses polling to efficiently wait for an event.
 #[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Default)]
 pub struct NonBlocking<'a> {
-    _marker: PhantomData<Context<'a>>,
+    /// The type `&'a &'a mut T` is invariant over `'a`, like `Context` is.
+    ///
+    /// We used to just use `Context` here, but then `Context` became `!Send`
+    /// and `!Sync`, making all of the futures that use this type `!Send` and
+    /// `!Sync` as well. So we just take the lifetime invariance and none of
+    /// the downsides.
+    _marker: PhantomData<&'a &'a mut ()>,
```

Comment:
> The type `&'a &'a mut T` is invariant over `'a`, like `Context` is.

`&'a mut T` makes `T` invariant, but `&'a T` doesn't. I think we need to use `&'a mut &'a ()` (instead of `&'a &'a mut ()`) to make `'a` invariant.

https://doc.rust-lang.org/nightly/nomicon/phantom-data.html#table-of-phantomdata-patterns


> Phantom type | variance of `'a` | variance of `T`
> -- | -- | --
> `PhantomData<&'a T>` | **co**variant | **co**variant
> `PhantomData<&'a mut T>` | **co**variant | **in**variant